cmake/Modules/ObsDefaults_macOS.cmake

```diff
@@ -0,0 +1,154 @@
+set(CMAKE_OSX_DEPLOYMENT_TARGET
+    "10.13"
+    CACHE STRING "OBS deployment target for macOS - 10.13+ required")
+if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+  set(CMAKE_INSTALL_PREFIX
+      "${CMAKE_BINARY_DIR}/rundir"
+      CACHE STRING "Directory to install OBS after building" FORCE)
+endif()
+
+# Enable modern cmake policies
+if(POLICY CMP0009)
+  cmake_policy(SET CMP0009 NEW)
+endif()
+
+if(POLICY CMP0011)
+  cmake_policy(SET CMP0011 NEW)
+endif()
+
+if(POLICY CMP0025)
+  cmake_policy(SET CMP0025 NEW)
+endif()
+
+macro(setup_obs_project)
+  # Set code-signing options
+  set(OBS_BUNDLE_CODESIGN_IDENTITY
+      "-"
+      CACHE STRING "OBS code signing identity for macOS")
+  set(OBS_CODESIGN_ENTITLEMENTS
+      "${CMAKE_SOURCE_DIR}/cmake/bundle/macOS/entitlements.plist"
+      CACHE INTERNAL "Path to codesign entitlements plist")
+  set(OBS_CODESIGN_LINKER
+      ON
+      CACHE BOOL "Enable linker code-signing on macOS (macOS 11+ required)")
+
+  # Xcode configuration
+  if(XCODE)
+    # Tell Xcode to pretend the linker signed binaries so that editing with
+    # install_name_tool preserves ad-hoc signatures. See CMake Issue 21854. This
+    # option is supported by codesign on macOS 11 or higher.
+    set(CMAKE_XCODE_GENERATE_SCHEME ON)
+    if(OBS_CODESIGN_LINKER)
+      set(CMAKE_XCODE_ATTRIBUTE_OTHER_CODE_SIGN_FLAGS "-o linker-signed")
+    endif()
+  endif()
+
+  # Set default options for bundling on macOS
+  set(CMAKE_MACOSX_RPATH ON)
+  set(CMAKE_SKIP_BUILD_RPATH OFF)
+  set(CMAKE_BUILD_WITH_INSTALL_RPATH OFF)
+  set(CMAKE_INSTALL_RPATH "@executable_path/../Frameworks/")
+  set(CMAKE_INSTALL_RPATH_USE_LINK_PATH OFF)
+
+  # Set bundle parameters for cmake's automatic plist generation
+  set(MACOSX_BUNDLE_EXECUTABLE_NAME "OBS")
+  set(MACOSX_BUNDLE_BUNDLE_NAME "${OBS_PRODUCT_NAME}")
+  set(MACOSX_BUNDLE_BUNDLE_VERSION "${OBS_BUILD_NUMBER}")
+  set(MACOSX_BUNDLE_COPYRIGHT "${OBS_LEGAL_COPYRIGHT}")
+  set(MACOSX_BUNDLE_GUI_IDENTIFIER "com.obsproject.obs-studio")
+  set(MACOSX_BUNDLE_ICON_FILE "AppIcon")
+  set(MACOSX_BUNDLE_SHORT_VERSION_STRING "${OBS_VERSION_CANONICAL}")
+
+  # Set paths for distribution bundling
+  set(OBS_BUNDLE_NAME "OBS")
+  set(OBS_EXECUTABLE_DESTINATION "${CMAKE_INSTALL_BINDIR}")
+  set(OBS_INCLUDE_DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/obs")
+  set(OBS_LIBRARY_DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+  set(OBS_CMAKE_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake")
+
+  # set(OBS_OUTPUT_LOCATION "${CMAKE_BINARY_DIR}/distribution")
+  # set(OBS_BUNDLE_LOCATION
+  # "${OBS_OUTPUT_LOCATION}/$<CONFIG>/${OBS_BUNDLE_NAME}.app")
+  # set(OBS_BUNDLE_BINARY_LOCATION "${OBS_BUNDLE_LOCATION}/Contents/MacOS")
+  # set(OBS_BUNDLE_FRAMEWORK_LOCATION
+  # "${OBS_BUNDLE_LOCATION}/Contents/Frameworks")
+  # set(OBS_BUNDLE_RESOURCE_LOCATION
+  # "${OBS_BUNDLE_LOCATION}/Contents/Resources") set(OBS_BUNDLE_PLUGIN_LOCATION
+  # "${OBS_BUNDLE_LOCATION}/Contents/PlugIns")
```

Comment:
If this isn't needed, it can be removed.

Reply:
Removed.